master/internal/rm/agentrm/resource_pool.go

```diff
@@ -479,8 +481,16 @@ func (rp *resourcePool) Receive(ctx *actor.Context) error {
 			defer func() {
 				rp.agentStatesCache = nil
 			}()
+
+			disallowedNodes := set.New[string]()
+			if msg.TaskID != nil {
+				disallowedNodes = *logpattern.DisallowedNodes(*msg.TaskID)
+			}
+			// check task speicifc slots here
 			for _, a := range rp.agentStatesCache {
-				totalSlots += len(a.slotStates)
+				if !disallowedNodes.Contains(a.Handler.Address().Local()) {
+					totalSlots += len(a.slotStates)
+				}
 			}
 		case rp.config.Provider.AWS != nil:
 			totalSlots = rp.config.Provider.MaxInstances * rp.config.Provider.AWS.SlotsPerInstance()
```

Comment:
could we somehow subtract the blocklist nodes from```totalSlots``` if the agents blocked are in the provided pool and been launched by the provider?

I'm not sure how hard this will be, if it is too hard I think it might be okay to not do